hazelcast/src/main/java/com/hazelcast/internal/blackbox/impl/FieldSensorInputFactories.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.blackbox.impl;
+
+import com.hazelcast.internal.blackbox.Counter;
+import com.hazelcast.internal.blackbox.DoubleSensorInput;
+import com.hazelcast.internal.blackbox.LongSensorInput;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+final class FieldSensorInputFactories {
+
+    private FieldSensorInputFactories() {
+    }
+
+    static void registerFieldTypes(Map<Class, FieldSensorInputFactory> factories) {
+        registerPrimitiveInt(factories);
+        registerPrimitiveLong(factories);
+        registerPrimitiveFloat(factories);
+        registerPrimitiveDouble(factories);
+        registerPrimitiveShort(factories);
+        registerAtomicLong(factories);
+        registerAtomicInteger(factories);
+        registerCounter(factories);
+        registerMapSize(factories);
+        registerCollectionSize(factories);
+    }
+
+    private static void registerMapSize(Map<Class, FieldSensorInputFactory> factories) {
+        factories.put(Map.class, new FieldSensorInputFactory() {
+            @Override
+            public LongSensorInput create(final Field field, final Object source) {
+                return new LongSensorInput() {
+                    @Override
+                    public long get() throws Exception {
+                        Map map = (Map) field.get(source);
```

Comment:
I realize here it doesn't matter, but the usage of a raw `Map` type will cause compiler warnings. `Map<?>` would work here (I think).